include/SDL3/SDL_process.h

```diff
@@ -0,0 +1,271 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+/**
+ * # CategoryProcess
+ *
+ * Process control support.
+ *
+ * These functions provide a cross-platform way to spawn and manage OS-level
+ * processes.
+ *
+ * You can create a new subprocess with SDL_CreateProcess(). If more complex
+ * functionality like redirecting input and output is necessary, you can use
+ * SDL_CreateProcessWithProperties().
+ *
+ * You can get the status of a created process with
+ * SDL_WaitProcess(), or forcibly terminate the process with
+ * SDL_KillProcess().
+ *
+ * Don't forget to call SDL_DestroyProcess() to clean up, whether the process
+ * process was killed, terminated on its own, or is still running!
+ */
+
+#ifndef SDL_process_h_
+#define SDL_process_h_
+
+#include <SDL3/SDL_error.h>
+
+#include <SDL3/SDL_begin_code.h>
+/* Set up for C function definitions, even when using C++ */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct SDL_Process SDL_Process;
+
+/**
+ * Create a new process.
+ *
+ * The path to the executable is supplied in args[0]. args[1..N] are additional arguments passed on the command line of the new process, and the argument list should be terminated with a NULL, e.g.:
+ *
+ * ```c
+ * const char *args[] = { "myprogram", "argument", NULL };
+ * ```
+ *
+ * See SDL_CreateProcessWithProperties() for more details.
+ *
+ * \param args the path and arguments for the new process.
+ * \param enable_IO SDL_TRUE to enable reading and writing to the process, SDL_FALSE to have the process output go to the application's output.
```

Comment:
I would call this something more like `pipe_stdio`, and I think it should be more precise about what it means. Perhaps something like:

`Setting pipe_stdio to SDL_TRUE is equivalent to setting SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN, SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN and SDL_PROP_PROCESS_REDIRECT_STDERR_BOOLEAN to true.`

`\param pipe_stdio SDL_TRUE to create pipes to the process's standard input and from the process's merged standard output and standard error; SDL_FALSE for the process to have no input and inherit the application's standard output and stdandard error.`

I'm not sure that reading from both standard output and standard error is a good default for a simplified API, though: it's a common convention to send machine-readable information to stdout, and human-readable unstructured diagnostic messages to stderr. If we merge stdout and stderr, the practical effect is that we'll corrupt the machine-readable format of what appears on stdout.

(For example, try running Steam's `steam-runtime-system-info` tool with `SRT_LOG=debug` in the environment. It sends valid JSON to stdout, and arbitrary human-readable noise which cannot be interpreted as JSON to stderr. If we mix the two up, the result cannot be machine-parsed.)

Reply:
Good point. I'll change it so stderr isn't redirected.